specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Patches.json

```diff
@@ -0,0 +1,380 @@
+      "post": {
+        "tags": [
+          "Patches"
+        ],
+        "operationId": "Patches_Apply",
+        "description": "Apply a PatchResource with patch id.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./Builders.json#/parameters/BuilderNameParameter"
+          },
+          {
+            "$ref": "#/parameters/PatchNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Patching operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/PatchResource"
+            }
+          },
+          "201": {
+            "description": "Patching operation has been started.",
+            "schema": {
+              "$ref": "#/definitions/PatchResource"
+            },
+            "headers": {
+              "Azure-AsyncOperation": {
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-examples": {
+          "Patches_Apply_0": {
+            "$ref": "./examples/Patches_Apply.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "PatchResource": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PatchProperties",
+          "description": "The resource-specific properties for this resource.",
+          "x-ms-client-flatten": true,
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        }
+      },
+      "description": "The resource representation of a patch.",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "ProvisioningState": {
+      "type": "string",
+      "description": "Resource instance provisioning state.",
+      "enum": [
+        "Succeeded",
+        "Failed",
+        "Canceled",
+        "Creating",
+        "Updating",
+        "Deleting"
+      ],
+      "x-ms-enum": {
+        "name": "ProvisioningState",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
```

Comment:
I think `ProvisioningState` is an interesting field to look at for the patch proxy resource -- as we discussed, there isn't necessarily a Container App Job execution spun up for a patch until the apply operation is executed for a given patch, so even though a new patch was detected and registered in the DB, what does that mean in terms of the `ProvisioningState` field?

I think the simple answer is that `Succeeded` represents "the concept of the patch being ready to be applied, but no resource has been created as a result", and `Deleting` represents "the concept of a patch being removed from the backend and the corresponding Container Apps Job execution being removed (if it isn't already done so when the patch is finished being applied)". Even writing this out, the simple answer is still pretty complicated, so I'm not sure if anyone has a better representation of this field and how we would even use it in the backend, if at all.

Reply:
Yep, I agree here for the purpose of the provisioning state field. Here is a more in depth explanation of it https://github.com/Azure/azure-resource-manager-rpc/blob/c71ccbd6a8adf4a47978e46bda1e7d405cb19b80/v1.0/async-api-reference.md#provisioningstate-property

Its mainly used so that customers can track the status of async operations, and then see the final state after the operation completes. Looks like the only async operation for us is going to be deletion, so I would probably have the 3 terminal states (succeeded, cancelled, failed) and then a (deleting) state. We also may want to think about what we discussed for making sure that the same patch that a customer deletes does not just turn around and get re-created. I imagine that this could have potential impacts on the API itself